test/test_exceptions.py

```diff
@@ -47,24 +48,29 @@ def test_gate_error_str_and_repr():
     str_expected = ("IonQGateError(\"gate 'a gate' is not supported on the 'b set' IonQ backends."
                     " Please use the qiskit.transpile method, manually rewrite to remove the gate,"
                     " or change the gateset selection as appropriate.\")"
-                   )
+                    )
     repr_expected = "IonQGateError(gate_name='a gate', gateset='b set')"
     assert str(err) == str_expected
     assert repr(err) == repr_expected
 
 
 def test_api_error():
     """Test that IonQAPIError has specific instance attributes."""
-    err = exceptions.IonQAPIError("an error", 500, "internal_error")
+    err = exceptions.IonQAPIError(
+        "an error", "http response", 500, "internal_error")
     assert err.message == "an error"
     assert err.status_code == 500
     assert err.error_type == "internal_error"
 
 
 def test_api_error_str_and_repr():
     """Test that IonQAPIError has a str/repr that includes args."""
-    err = exceptions.IonQAPIError("an error", 500, "internal_error")
-    expected = "IonQAPIError(message='an error',status_code=500,error_type='internal_error')"
+    err = exceptions.IonQAPIError(
+        "an error", "http response", 500, "internal_error")
```

Comment:
can you try this with a real requests.Response to see if it serializes OK?

Reply:
alright im now using requests.request and response so it's pretty realistic